tensorflow/compiler/xla/python/pytree.cc

```diff
@@ -491,7 +491,8 @@ py::list PyTreeDef::FlattenUpTo(py::handle xs) const {
 }
 
 py::object PyTreeDef::Walk(const py::function& f_node, py::handle f_leaf,
-                           py::iterable leaves) const {
+                           py::iterable leaves,
+                           bool pass_node_data) const {
```

Comment:
Given this isn't an API we advertise or document, I might be tempted just to change the API contract of f_node to unconditionally pass the node data.

The only use of this function I can find anywhere is this one: https://github.com/google/jax/blob/be41c8c1d3ab79078acd9113f35e179a4ebf8f33/jax/_src/tree_util.py#L195
and as best I can tell, that function is completely unused. So I might just delete it.

Reply:
Thanks! That was my first design, so I'll happily revert to it.  I wasn't sure if there might be keen `Walk`ers in other parts of the ecosystem.